src/vfs2.c

```diff
@@ -420,15 +452,10 @@ static void write_basic_hdr_cksums(struct wal_index_basic_hdr *bhdr)
  */
 static int shm_init(struct shm *shm, struct wal_hdr whdr)
 {
-	shm->regions = sqlite3_malloc64(VFS2_WAL_INDEX_REGION_SIZE);
-	if (shm->regions == NULL) {
+	struct shm_region *r0 = shm_grow(shm);
```

Comment:
Maybe we should initialize shm here to zero just in case the caller is using uninitialized memory and that leads to bugs, I cannot think of any use case that requires setting something by hand.

BTW: another custom vector type we just created.

Reply:
> Maybe we should initialize shm here to zero just in case the caller is using uninitialized memory and that leads to bugs, I cannot think of any use case that requires setting something by hand.

I don't think this is true of any of the callsites right now, but better safe than sorry. I'll add the defensive initialization.

Reply:
Looking at this again, I think maybe it would make more sense to have an invariant for the shm state machine that asserts that the other fields are zeroed when the state is SHM_EMPTY. Defensively zeroing them here would suggest that we expect the struct to be filled with garbage, but that's not the case, we expect it to be "initialized as empty" already. (Maybe `shm_init` is not the best name for that reason.)